src/server.c

```diff
@@ -4336,14 +4336,19 @@ int finishShutdown(void) {
     /* Close the listening sockets. Apparently this allows faster restarts. */
     closeListeningSockets(1);
 
+    if (server.cluster_enabled) {
+        /* The error logs have been logged in the save function if the save fails. */
+        serverLog(LL_NOTICE, "Saving the cluster configuration file before exiting.");
+        clusterSaveConfig(1);
+    }
```

Comment:
Should this be a direct `clusterSaveConfig` invocation or should a generic method `clusterHandleServerShutdown` like invoked? Depending on the cluster implementation further action should be taken. For `cluster_legacy.c` we can call `clusterSaveConfig(1)` inside it.

WDYT ?

Reply:
good idea, i updated the code.